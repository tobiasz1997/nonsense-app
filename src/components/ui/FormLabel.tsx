import { FC, PropsWithChildren } from 'react';

type Props = {
	required?: boolean;
};

const FormLabel: FC<PropsWithChildren & Props> = (props) => {
	return (
		<span className="flex space-x-1 text-xs">
			<span className="block font-bold">{props.children}</span>
			{props.required && <span className="text-red">*</span>}
		</span>
	);
};

export default FormLabel;
