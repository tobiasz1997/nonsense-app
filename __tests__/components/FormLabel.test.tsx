/*
 * @jest-environment jsdom
 * */

import { render, screen } from '@testing-library/react';
import FormLabel from '@components/ui/FormLabel';

describe('Form Label', () => {
	const content = '@#$%^&*()qwerty';

	it('should render without crashing', () => {
		const { container } = render(<FormLabel>{content}</FormLabel>);
		expect(container).toMatchSnapshot();
	});

	it('should value be correct', () => {
		render(<FormLabel required>{content}</FormLabel>);

		const text = screen.getByText(content);

		expect(text).toBeInTheDocument();
	});
});
