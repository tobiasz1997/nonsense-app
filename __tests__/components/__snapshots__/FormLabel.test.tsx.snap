// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Label should render without crashing 1`] = `
<div>
  <span
    class="flex space-x-1 text-xs"
  >
    <span
      class="block font-bold"
    >
      @#$%^&*()qwerty
    </span>
  </span>
</div>
`;
